pkg/apis/hive/v1/clusterpool_types.go

```diff
@@ -62,6 +62,11 @@ type ClusterPoolSpec struct {
 	// SkipMachinePools allows creating clusterpools where the machinepools are not managed by hive after cluster creation
 	// +optional
 	SkipMachinePools bool `json:"skipMachinePools,omitempty"`
+	// ClaimLifetime is the maximum lifetime of the claim after it is assigned a cluster. If the claim still exists
+	// when the lifetime has elapsed, the claim will be deleted by Hive.
+	// The lifetime of a claim is the mimimum of the lifetimes set by the cluster pool and the claim itself.
+	// +optional
```

Comment:
I'm concerned about one use case here, if someone were to hit a serious issue and needed to keep a cluster around as a reproducer until someone could investigate. With this policy I don't think there's any way for them to override that and it's the kind of thing that seems like it almost surely will arise at some point.

We should probably assume some level of trust and let explicit lifetime requests on a claim be respected, even if it's just on updates. We could use this logic on create, but let users explicitly modify it after creation forcing them to take a more explicit step. Tooling to look for clusters around too long would be easy to write.

Reply:
i think goal for setting a lifetime by the clusterpool admin is to make sure no cluster stays up _for long_ because these clusters use up the resources/quota for the clusterpool.  only setting the lifetime at create and allowing **users** to override the lifetime to anything seems incorrect from the admin's perspective as users should not be allowed to override the will the admin.

> 'm concerned about one use case here, if someone were to hit a serious issue and needed to keep a cluster around

i don't think user that created the claim should have the power to decide or lengthen the lifetime of the cluster and it should only be the admins that have the power to make the exception. 

the admins could detach the clusterdeployment from the pool/claim so that it lives for longer.. _not sure yet if that is possible_

Reply:
My assumption was it was best as a guideline from the admin to make sure nobody forgot to set a lifetime, and they would be willing to exercise a level of trust. I'll ping Steve and some ACM folks and get some feedback and see.